src/freenas/usr/local/libexec/nas/generate_smb4_conf.py

```diff
@@ -1627,6 +1627,25 @@ def restore_secrets_database():
     smb4_restore_tdbfile(backup, secrets)
 
 
+def smb4_do_migrations(client):
+    sentinel_directory = "/data/sentinels/samba"
+
+    if not os.access(sentinel_directory, os.F_OK):
+        smb4_mkdir(sentinel_directory)
+        os.chmod(sentinel_directory, 0o700)
+
+    # 11.1-U3 -> 11.1-U4
+    def migrate_11_1_U3_to_11_1_U4(client):
+        samba_user_import_file = "/var/db/samba4/.usersimported"
```

Comment:
@jhixson74  Is it necessary to check we are doing this only for FreeNAS?

Reply:
Nope. Since we are also doing a migration here for TrueNAS (perhaps we should have separate method for TrueNAS only migrations?), this will work as expected. FreeNAS is migrating from /root/samba/private/* to /var/db/samba4/private/, where as TrueNAS is migrating from /var/db/samba4/private/* to /root/samba/private/ ;-)

Reply:
So by nuking the .usersimported file, we regenerate the tdb files no matter where the private dir is set too

Reply:
Makes sense then!